source/Calamari/Integration/FileSystem/PackageStore.cs

```diff
@@ -108,57 +72,56 @@ public StoredPackage GetPackage(string prefix, PackageMetadata metadata)
 
         public IEnumerable<StoredPackage> GetNearestPackages(string packageId, SemanticVersion version, int take = 5)
         {
-            var root = GetPackageRoot(null);
-            fileSystem.EnsureDirectoryExists(root);
+            fileSystem.EnsureDirectoryExists(rootDirectory);
 
-            return (
-                from file in fileSystem.EnumerateFilesRecursively(root, packageId + "*.nupkg-*")
-                let package = ReadPackageFile(file)
+            var zipPackages =
+                from filePath in fileSystem.EnumerateFilesRecursively(rootDirectory, packageId + "*.nupkg-*")
+                let zip = ReadZipPackage(filePath)
+                where zip != null && zip.Id == packageId && zip.Version < version
```

Comment:
While in some sense its technically correct, it looks like the way this is used is it checks for exact match first (using `GetPackage` method), and then assuming no match it then does this call to find the _nearest_ match. This explains why no-one has brought this up as a bug before.